catroid/src/main/java/org/catrobat/catroid/content/actions/AskSpeechAction.kt

```diff
@@ -0,0 +1,116 @@
+import android.speech.RecognizerIntent
+import android.util.Log
+import com.badlogic.gdx.scenes.scene2d.Action
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+import org.catrobat.catroid.formulaeditor.UserVariable
+import org.catrobat.catroid.stage.StageActivity
+import org.catrobat.catroid.stage.StageActivity.IntentListener
+import java.util.ArrayList
+
+class AskSpeechAction : Action(), IntentListener {
+
+    var sprite: Sprite? = null
+    var questionFormula: Formula? = null
+    var answerVariable: UserVariable? = null
+    private var questionAsked = false
+    private var answerReceived = false
+
+    private fun askQuestion() {
+        StageActivity.messageHandler?.obtainMessage(StageActivity.REGISTER_INTENT,
+            listOf(this))?.sendToTarget()
+
+        questionAsked = true
+    }
+
+    private fun createRecognitionIntent(question: String?) =
+        Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH).also { intent ->
+            intent.putExtra(
+                RecognizerIntent.EXTRA_LANGUAGE_MODEL,
+                RecognizerIntent.LANGUAGE_MODEL_FREE_FORM
+            )
+
+            if (question != null && question.length != 0) {
+                intent.putExtra(RecognizerIntent.EXTRA_PROMPT, question)
+            }
+        }
+
+    fun setAnswerText(answer: String?) {
+        answerVariable?.apply {
+            value = answer
+        }
+
+        answerReceived = true
+    }
+
+    override fun act(delta: Float): Boolean {
+        if (!questionAsked) {
+            askQuestion()
+        }
+        return answerReceived
+    }
+
+    override fun restart() {
+        questionAsked = false
+        answerReceived = false
+        super.restart()
+    }
+
+    override fun getTargetIntent(): Intent {
+        val question = try {
+            questionFormula?.interpretString(sprite) ?: ""
+        } catch (e: InterpretationException) {
+            Log.e(TAG, "Formula interpretation in ask brick failed")
+            ""
+        }
+        return createRecognitionIntent(question)
+    }
+
+    override fun onIntentResult(resultCode: Int, data: Intent) {
+        val matches: ArrayList<String>?
+        when (resultCode) {
+            Activity.RESULT_OK -> {
+                matches = data.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)
+                Log.d(TAG, "Results Speechrecognition: $matches")
+                if (matches != null && matches.size > 0) {
+                    setAnswerText(matches[0])
+                } else {
+                    setAnswerText("")
+                }
+            }
+            Activity.RESULT_CANCELED, Activity.RESULT_FIRST_USER -> setAnswerText("") // User canceled action
+            else -> Log.e(TAG, "unhandled speech recognizer resultCode $resultCode")
+        }
+    }
+
+    companion object {
+        private const val TAG = "AskSpeechAction"
```

Comment:
you can use AskSpeechAction::class.java.simpleName instead of hard coding

Reply:
Just `javaClass.simpleName` also works